### PR TITLE
tests: s/GITEA_UNIT_TESTS_VERBOSE/GITEA_UNIT_TESTS_LOG_SQL/

### PR DESCRIPTION
**Pull request by singuliere**
The GITEA_UNIT_TESTS_VERBOSE variable is an undocumented variable
introduced in 2017 (see 1028ef2defd94a64f2433b07fe5d93681864cebb)
whose sole purpose has been to log SQL statements when running unit
tests.

It is renamed for clarity and a warning is displayed for backward
compatibility for people and scripts that know about it.

The documentation is updated to reflect this change.

### CODE REVIEW

**Review by zeripath — APPROVED** [review]
I think renaming this undocumented variable is reasonable.

**Comment by wxiaoguang** [comment]
After renaming, I think we can safely drop the `GITEA_UNIT_TESTS_VERBOSE` together in this PR: it was undocumented, and only for developers, it's safe to be removed. IMO no one will be hurt or unhappy  :D

**Comment by zeripath** [comment]
Agreed either way

**Comment by singuliere** [comment]
> After renaming, I think we can safely drop the `GITEA_UNIT_TESTS_VERBOSE` together in this PR: it was undocumented, and only for developers, it's safe to be removed. IMO no one will be hurt or unhappy :D

I dropped the backward compatibility change as requested.

**Review by 6543 — APPROVED** [review]

File: models/unittest/testdb.go [hunk 1]
```diff
@@ -14,6 +14,7 @@ import (
 
 	"code.gitea.io/gitea/models/db"
 	"code.gitea.io/gitea/modules/base"
+	"code.gitea.io/gitea/modules/log"
```
Comment (6543) [inline models/unittest/testdb.go]:
```suggestion
```

**Review by wxiaoguang — APPROVED** [review]

**Pull request merged**